app/controllers/api/v1/products_controller.rb

```diff
@@ -37,6 +37,14 @@ def destroy
         head :no_content
       end
 
+      def assign_product_details
+        if @product.update(product_details: [product_params[:product_details].first])
```

Comment:
Ticket TS-708
The endpoint should NOT delete the currently assigned product_details.

The endpoint is not deleting the currently assigned product_details but it is overriding the currently assigend product_details.